cunumeric/array.py

```diff
@@ -2185,6 +2185,98 @@ def dumps(self):
         """
         return self.__array__().dumps()
 
+    def _normalize_axes_shape(self, axes, s):
+        user_axes  = axes is not None
+        user_sizes = s is not None
+        if user_axes and user_sizes and len(axes) != len(s):
+            raise ValueError(
+                "Shape and axes have different lengths"
+                )
+        if user_axes:
+            fft_axes = axes
+        else:
+            fft_axes = range(-len(s),0) if user_sizes else range(self.ndim)
+        if np.max(np.abs(list(fft_axes))) > self.ndim:
+            raise ValueError(
+                "Axis is out of bounds for array of size {}".format(self.ndim)
+            )
+        fft_axes = [x % self.ndim for x in fft_axes]
```

Comment:
Or maybe this could be before the check above.

Reply:
No, because like you pointed out in a previous comment, if this goes before the check, then
`axis=-5` or `axis=7` would be accepted, when they should not.

i.e. valid axes for a 3D FFT should be within [-2, ..., 2]